lib/rubocop/cop/badge.rb

```diff
@@ -10,37 +10,23 @@ module Cop
     # allow for badge references in source files that omit the department for
     # RuboCop to infer.
     class Badge
-      # Error raised when a badge parse fails.
-      class InvalidBadge < Error
-        MSG = 'Invalid badge %<badge>p. ' \
-              'Expected `Department/CopName` or `CopName`.'
-
-        def initialize(token)
-          super(format(MSG, badge: token))
-        end
-      end
-
       attr_reader :department, :cop_name
 
       def self.for(class_name)
-        new(*class_name.split('::').last(2))
+        parts = class_name.split('::')
+        parts_without_first_two = parts[2..-1]
```

Comment:
Could we come up with a better name for this variable?

Actually, I wonder if what we are _really_ checking is whether the class name is a subclass of `RuboCop::Cop`/`RuboCop::Base`? In that case, I’d suggest a more explicit namespace check:
```rb
parts = class_name.split('::')
if class_name.start_with?('RuboCop::Cop', 'RuboCop::Base')
  new(parts[2..-1])
else
  new(parts.last(2))
end
```

Reply:
Thank could also be `MyCoolCompany::Cop::Mailers::ExplicitRecipient`.

Reply:
How about e.g. `MyCoolCompanyCops::Mailers::ExplicitRecipient`? It seems a bit arbitrary to decide that the first *two* namespaces are not part of the department name.

Even worse, how about `MyCoolCompany::Cop::Style::Alias`? Should we would allow people’s internal cops to conflict with our cops.

Reply:
Also, as an experiment, I've moved one of the `rubocop-rspec` cops up the hierarchy, and removed its department altogether. Everything went quite smoothly.
I'm not aware if the department is mandatory for cops, but if it worked previously, I believe it should continue to work.

Reply:
> Could we come up with a better name for this variable?

Actually we can get rid of this variable. I'm gonna update the PR.

> Actually, I wonder if what we are really checking is whether the class name is a subclass of
> RuboCop::Cop/RuboCop::Base? In that case, I’d suggest a more explicit namespace check:

Yes, if we are to change the code to the one proposed by @bquorning above, then we would still possibly have the same problem for cops `MyCoolCompany::Cop::Mailers::ExplicitRecipient` and `MyCoolCompany::Cop::Rails::Mailers::ExplicitRecipient`.

> It seems a bit arbitrary to decide that the first two namespaces are not part of the department name.

I do understand that it is arbitrary to decide the first two namespaces are not part of the department name but I don't see a better solution right now, and this particular solution I'm proposing fixes the problem for all the current cases I think.

> Even worse, how about MyCoolCompany::Cop::Style::Alias? Should we would allow people’s internal cops to 
> conflict with our cops.

That's a problem. But maybe a different one? Once and for all this can be fixed only if you put every part of this cop class name into its department/class name, something like `MyCoolCompany/Cop/Style/Alias`. It would require everyone to change every existing cop. I am not happy proposing such a huge change :) Can this be fixed another way?